### PR TITLE
ConsumerBasedExpectedCheckVerifier: FIXUP issue where verifyNotNull always passed.

### PR DESCRIPTION
**Pull request by tsmock**
Also, fix some sonar lint issues.

### Description:

The verifier added in `verifyNotNull` is only ever run against `generatedFlags`, which is a final field which is instantiated in the constructor for `ConsumerBasedExpectedCheckVerifier`, which means that `verifyNotNull` is essentially a no-op.

Primary change:
* Keep the current `assertNotNull` of the list, but afterwards do a `forEach` loop on the list checking for null `CheckFlag` objects.

Secondary changes:
* `flags.size() == 0` -> `flags.isEmpty()`
* `assertTrue(flags.size() > 0)` -> `assertFalse(flags.isEmpty())`
* `throw new Exception` -> `Assert.fail`

### Potential Impact:

Any downstream user of `ConsumerBasedExpectedCheckVerifier` may see new failing tests, since `verifyNotNull` is no longer a no-op.

### Unit Test Approach:

N/A

### Test Results:

No current tests fail.

### CODE REVIEW

**Comment by tsmock** [comment]
> SonarCloud Quality Gate failed.
> 
> [Bug](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=BUG) [C](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=BUG) [1 Bug](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=VULNERABILITY) (and [Security Hotspot](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=SECURITY_HOTSPOT) to review)
> [Code Smell](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&metric=new_coverage&view=list) [87.5% Coverage](https://sonarcloud.io/component_measures?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&metric=new_coverage&view=list)
> [](https://sonarcloud.io/component_measures?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=org.openstreetmap.atlas%3Aatlas-checks&pullRequest=414&metric=new_duplicated_lines_density&view=list)

It looks like the "bug" is likely an issue with sonar lint not recognizing that `Verifier` should be a valid location for assertions (the class extends `org.junit.rules.Verifier`, and the sonar bug is "Remove this assertion from production code"). I'll look into this more, but I think I'll find one of two things:
1) No assertions are expected in JUnit Verifiers/AfterEachCallback, even though it is non-production code
2) Sonar does not see `org.junit.rules.Verifier` as non-production code and needs to be fixed

**Comment by Bentleysb** [comment]
@tsmock, I agree with your assessment of the sonar report. It looks like we have not updated anything in the this class since sonar was added to the project. Feel free to just suppress this warning.

**Comment by tsmock** [comment]
I've done a bit more digging, and part of the problem is that the Verifiers are not in the actual test code -- they are in the main source code (`src/main/java`) instead of the test code (`/src/test/java`). TBH, I still think it is a bug in sonar (the error should be "Test class not in test package"), but that is a different problem.

I've pushed a commit that moves both verifiers into the test sources.

**Comment by Bentleysb** [comment]
The reason this code is not currently in the test package is because we want it to be available for downstream projects. Lets keep these classes where they are and just suppress the sonar warning.  You should be able to do this by adding `@SuppressWarnings(squid:S5960)`.

**Comment by tsmock** [comment]
Fair enough. I'll move them back and add some information to the javadocs so that a future person does not move them.

**Comment by Bentleysb** [comment]
The build is failing due to spotless linting issues.

**Comment by tsmock** [comment]
> The build is failing due to spotless linting issues.

My bad. I usually run `./gradlew spotlessApply` prior to committing, but I didn't do it this time. I need to add a pre-commit hook.

**Review by Bentleysb — APPROVED** [review]

File: src/main/java/org/openstreetmap/atlas/checks/validation/verifier/ConsumerBasedExpectedCheckVerifier.java [hunk 1]
```diff
@@ -125,7 +124,11 @@ public ConsumerBasedExpectedCheckVerifier verifyNotEmpty()
      */
     public ConsumerBasedExpectedCheckVerifier verifyNotNull()
     {
-        this.globalVerifiers.add((final List<CheckFlag> flags) -> Assert.assertNotNull(flags));
+        this.globalVerifiers.add((final List<CheckFlag> flags) ->
+        {
+            Assert.assertNotNull(flags);
+            flags.forEach(Assert::assertNotNull);
```
Comment (seancoulter) [inline src/main/java/org/openstreetmap/atlas/checks/validation/verifier/ConsumerBasedExpectedCheckVerifier.java:130]: Would be an issue if we even had a null CheckFlag!

**Review by seancoulter — APPROVED** [review]

**Pull request merged**